onnxruntime/core/common/cpuid_info.cc

```diff
@@ -134,6 +135,24 @@ void CPUIDInfo::X86Init() {
         // avx512_skylake = avx512f | avx512vl | avx512cd | avx512bw | avx512dq
         has_avx512_skylake_ = has_avx512 && (data[1] & ((1 << 16) | (1 << 17) | (1 << 28) | (1 << 30) | (1 << 31)));
         is_hybrid_ = (data[3] & (1 << 15));
+        // Check for TPAUSE
+        CheckIntelResult check_intel = CheckIntel();
+        if (check_intel.is_intel) {
+          if (data[2] & (1 << 5)) {
+            has_tpause_ = true;
+          } 
+          else {
+            has_tpause_ = false;
+          }
+
+          #ifdef __linux__
+          if(__builtin_cpu_supports("waitpkg")) {
+            has_tpause_ = true;
+          } else {
+            has_tpause_ = false;
+          }
+          #endif
```

Comment:
the `#ifdef __linux__` part overwrites the previous value of `has_tpause_`. could we simplify it to this?

```suggestion
#ifdef __linux__
          has_tpause_ = __builtin_cpu_supports("waitpkg") != 0;
#else
          has_tpause_ = (data[2] & (1 << 5)) != 0;
#endif
```